### DISCUSSION TITLE
External flash dump

### DISCUSSION THREAD

**Discussion by pop3556** [pcmihnea/HA_WMR500#10] [Q&A]
Hello,

Great thanks to you for sharing your work.

The dump of your firmware have  allow to me to change my stm32 that had its flash corrupt and reprogram it. it have allow done  working communication with sensor and datas display.

Unfortunately, my wifi won't work. I had trying to spy wifi communications with kismet and airodump-ng or try to connect to SSID OS_WMR500_ password 12345678 with a computer, nothing happened...

I suspect a corrupt external flash, in this flash there is parameters (begins at 0x0 and 0x4000), copy of the firmware(begin at 0x64000) and an area with other software (begin at 0x9000) that I can't check without an other reference file.

So my question is ... Have you a dump of your external flash that you can share ? it' could help my investigations... Thanks.

**Comment by pcmihnea** [comment]
Hi,

Good to know my project is of help to you!

Odd that both chips failed - was is from a power issue? Or water ingress 😄 ? Is there a possibility the WiFi chip may also be affected?
Got any useful log from the debug UART?

I do recall the user configuration (WiFi credentials, GUID, units) and measurements history/statistics are also present in the external flash.
Not sure about CRC, as for example changing the WiFi config seem to affects only the SSID/password values stored plain-text at 0x1AA2 and 0x1ADA respectively. 0x229 to 0x271, and 0x4229 to 0x4271 contain the GUID.

**Reply by Cla92240** [reply]
Hi, 
I face the same issue (wifi OS_WMR500C_xxxx doesn't appear any more when I try a factory reset.
Did you solve your problem?
Thanks in advance

**Reply by pcmihnea** [reply]
Did you try entering WiFi pairing mode again?

**Reply by Cla92240** [reply]
Yes I tried many times. I even tried to connect manually to the "OS_WMR500C_xxxx".
But I didn't succeed
Thanks in advance for your advices

**Comment by pop3556** [comment]
Hi !

I have seen your message by Email notification ... There is a long time that I didn't connected...

So firstly, I had not taken the time to thanks pcmihnea for it dump sharing ! So thanks for you reactivity. I been able to confirm that my dump add differents from yours but it was not the issue...

To repair my station, I had firstly change the STM32 (STM32F411) because it was dead, it was my first problem due to I think microcuttings on the electrical network. After this the wifi were not working. After lot of analyses, I found that my Wifi issue was coming from an issue on the new STM32 (a bad IO or a bad solder, I don't know, on one of the 4 SDIOs bus pins linked to the WIFI component). I found it by do and check communication between STM32 and the WIFI component (Wifi component : AMPAK AP6212 CYW43438) (a bit state was always missing on the data's frames on the 4 bits bus's communication ...).
The SDIO is a 26Mhz bus and the issue was appear only when the STM32 communicate with the WIFI component. When I was check state HIGH/LOW on SDIO pins on the STM32 (on DC mode, with a multimeter) the level were goods. So I don't know if it was due to bad solder or bad STM32 component (maybe an ESD issue ...).

So to conclude I have change and resolder a third STM32 a second time and it have solved the problem ...

After That I have try to follow the pcmihnea topic, some parts was not easy for a none expert ! )


I join, to this respond, a script that I create for my personal consumption, it's not an expert script but it allow to connect to the station, set time and communicate with it. It working on Debian bullseye for example with a Wifi card that allow making Hotpost (I use it on a portable computer). I'm not responsable of what could append on your computer after launch it, I just share it if it can help. Of course you must adjust logins, ID password to your using (I have put bad values). 7zip file with password "WMR500C".

For your information, You can manually set the hour with the buttons, I not found documents with this information... (It's written in my script)

Regarding the Wifi pairing, Download the "Mobile telnet" app on google play,
On the device,
    $Device_ID_4_firsts_characters=    Get Device ID (beginning of the GUUID) by pressing long the device buttons "Select"+"Arrow Up"  
Press short the PAIR button,
Press long the PAIR button,
Press short the PAIR button (to select wifi),
Connect wifi of the phone to 
    SSID : OS_WMR500C_$Device_ID_4_firsts_characters
    Password : 12345678
To Android 'Telnet Mobile' app for example,
Connect to 192.168.10.1 port 50007
send WMR500C($Hotpost_SSID_Len$Hotpost_SSID,$Hotpost_Password_Len$Hotpost_Password),
$Hotpost_SSID=MY__SSID_HOTPOST    #your hotpost ssid
$Hotpost_Password=password               #your hotpost  password
$Hotpost_SSID_Len=16                      #in decimal
$Hotpost_Password_Len=08                  #in decimal
The device must respond with the device GUUID on the telnet app
send CONFIRM
>>>Normaly the device is configured


pcmihnea an information is missing on your topic, regarding HTTP ports :
the port set on "wmr500_1490_patched_generic.bin" is 50007,
the port set on "wmr500_1490_patched_appdaemon.bin" is "5050",
the port set on "wmr500_1490_original.bin" is 443.


Thanks for all.


Regards.
[Start_WMR500C_Servers__20230714.zip](https://github.com/pcmihnea/HA_WMR500/files/12448410/Start_WMR500C_Servers__20230714.zip)
